### PR TITLE
BUG: Past enrollments not displaying on edit enrollments page

### PR DESCRIPTION
**Pull request by nickbristow**
## Background
On roster redesign, past enrollments not displaying on child edit enrollments page

## GitHub Issue
#1555 

## Associated PRs
If applicable, replace this text with links to additional pull requests associated with this work.

## Validation Plan
Please outline the steps you've already taken to validate this body of work, along with any additional steps that need to be taken in order to complete testing.

## Tech Specs
- [ ] If any TypeORM entities were created/updated/removed, the necessary migration(s) were [correctly auto-generated](https://github.com/typeorm/typeorm/blob/master/docs/using-cli.md#generate-a-migration-from-existing-table-schema) and **not manually written**.

## Automated Testing
- [ ] All unit tests are passing.
- [ ] Unit tests have been added to cover this changeset.

## Additional Context
If there's any additional information that those reviewing or validating this work should be aware of, please specify it here.

### CODE REVIEW

**Comment by jessethach** [comment]
@nickbristow This might actually implicitly resolve #1554. I had a similar solution in my PR #1562. I'm going to pull the branch and see if that's the case.

**Comment by jessethach** [comment]
> @nickbristow This might actually implicitly resolve #1554. I had a similar solution in my PR #1562. I'm going to pull the branch and see if that's the case.

Pulled down the code and confirmed that bug #1554 is not resolved by these changes.

File: src/controllers/children/get.ts [hunk 1]
```diff
@@ -76,7 +82,9 @@ const queryBuilderBuilder = async ({
     'Child.enrollments',
     'enrollment',
     // we expect the active enrollment to not have an exit or deleted date
-    'enrollment.exit IS NULL AND enrollment.deletedDate IS NULL'
+    pastEnrollments
```
Comment (jessethach) [inline src/controllers/children/get.ts:85]: Is there a particular reason to not be using `andWhere` chaining methods as opposed to string queries?

**Review by jessethach — APPROVED** [review]
Just a comment and confirmation of another bug persisting on this branch!

File: src/controllers/children/get.ts (hunk 1 above)
Comment (nickbristow) [reply src/controllers/children/get.ts:85]: Nope! i just modified the code enough to insert my switch in there.

**Review by JordanGuinn — APPROVED** [review]

**Comment by JordanGuinn** [comment]
LGTM! Though we are probably gonna wanna unify on how we handle the retrieval of previous enrollments, as the [withdraw bugfix PR](https://github.com/ctoec/data-collection/pull/1562) has something very similar implemented slightly differently

**Review by jessethach — APPROVED** [review]

**Review by lukenewman — APPROVED** [review]

**Comment by bamader** [comment]
## Validation Plan
Upload a file to the roster to get some children loaded. Ensure that this file contains at least one child with a historical enrollment. Then:

- [ ] Ensure that the child with the historical enrollment in the upload displays both enrollments in the Edit Record page
- [ ] Edit a child's enrollment/funding information and ensure the change is persisted
- [ ] Change a child's enrollment using the change enrollment button/modal, and ensure the change persists and that the previous enrollment appears
- [ ] Verify ability to edit previous enrollments

**Comment by bamader** [comment]
Validations are okay for using the buttons in the app and keeping things to 2 enrollments, but for uploads or more than 2 previous enrollments, things get a little buggy. You can see some of the ways detailed in the validation comment above. 

I also found a bug regarding children with multiple enrollments and how they're counted. It looks like the number of children reported at the top of the roster under the org heading is incorrectly based on the total number of enrollments in the respective roster, rather than the number of children. See image:
Screen Shot 2021-11-17 at 1 43 19 PM
To produce this, I uploaded this spreadsheet: 
[ECE Data Collection Template (v3).xlsx](https://github.com/ctoec/data-collection/files/7557190/ECE.Data.Collection.Template.v3.xlsx)
which contains twelve records (one of the children has a historical enrollment). We should align these counts so that the proper information is reported (potentially related note, for the part of the roster subheading that says 12 enrollments, I might suggest changing this to 12 active enrollments? Maybe not a big deal, but potentially worthwhile given that we're now broadening the scope of the roster to show historical enrollments).

**Comment by jessethach** [comment]
## Validation Plan
Upload a file to the roster to get some children loaded. Ensure that this file contains at least one child with a historical enrollment. Then:

---

- [ ] Ensure that the child with the historical enrollment in the upload displays both enrollments in the Edit Record page

Screen Shot 2021-11-17 at 10 44 44 AM

---

- [ ] Edit a child's enrollment/funding information and ensure the change is persisted

https://user-images.githubusercontent.com/10701395/142264044-98403a57-5d40-472d-87d6-b9d6349ebc5a.mov

---

- [ ] Change a child's enrollment using the change enrollment button/modal, and ensure the change persists and that the previous enrollment appears

https://user-images.githubusercontent.com/10701395/142264606-053d0c95-a3c5-47f0-af03-5a76f37926ae.mov

---

- [ ] Verify ability to edit previous enrollments

https://user-images.githubusercontent.com/10701395/142265314-127f23e7-c514-483d-851b-326f620b6b4d.mov

@nickbristow I am not sure if the validation occurring in this video is expected. IF it is, we can check it complete!

**Comment by nickbristow** [comment]
@bamader updated to order enrollments when there are multiple without end dates

**Comment by bamader** [comment]
@nickbristow Finished re-validation, and the sorting and multiple enrollments seem to be fixed. However, I found another problem. When you either upload a sheet containing a record with multiple enrollments, or you change a child's enrollment using the Edit Record buttons, an additional "-1" is applied to the day value of both enrollment start and the funding start (as well as corresponding exits).

Repro steps:
- from an empty roster, upload this sheet 
[ECE Data Collection Template (v3).xlsx](https://github.com/ctoec/data-collection/files/7565041/ECE.Data.Collection.Template.v3.xlsx)
- click on Orlando Conroy and go to his enrollment/funding tab
- Click change enrollment
- Give him enrollment start and funding start dates of 06/06/2021
- Save the new enrollment, and observe that they're saved/displayed as 06/05/2021

**Comment by bamader** [comment]
@nickbristow How we doing on this ticket? Anything I can help polish up or investigate?

**Comment by nickbristow** [comment]
> @nickbristow Finished re-validation, and the sorting and multiple enrollments seem to be fixed. However, I found another problem. When you either upload a sheet containing a record with multiple enrollments, or you change a child's enrollment using the Edit Record buttons, an additional "-1" is applied to the day value of both enrollment start and the funding start (as well as corresponding exits).
> 
> Repro steps:
> 
> * from an empty roster, upload this sheet
>   [ECE Data Collection Template (v3).xlsx](https://github.com/ctoec/data-collection/files/7565041/ECE.Data.Collection.Template.v3.xlsx)
> * click on Orlando Conroy and go to his enrollment/funding tab
> * Click change enrollment
> * Give him enrollment start and funding start dates of 06/06/2021
> * Save the new enrollment, and observe that they're saved/displayed as 06/05/2021

This appears to not be related to this PR and will be addressed in a separate ticket.

**Comment by bamader** [comment]
With the opening of another ticket for the date subtraction bug, this is good to go

**Pull request merged**